src/common/utils.cpp

```diff
@@ -306,6 +309,81 @@ std::string toUpper(std::string_view s)
 }
 std::string toUpper(const std::string& s) { return toUpper(std::string_view(s)); }
 
+#ifndef WIN32
+
+// Resolve case-insensitive path on case-sensitive file systems.
+// This does IO.
+// Preserves non-existing paths. Does not trim redundant `./`.
+// Length out of the returned string is the same as of the input string.
+//
+// This function assumes paths without drive letter assignments and
+// operates on strings directly.
+//
+// Example:
+// On your system there is a `/tmp/AFileWithUpperCaseInIt`.
+// `resolvePathCaseInsensitively("/tmp/afilewithuppercaseinit")` =
+// `"/tmp/AFileWithUpperCaseInIt"`
+static std::string resolveCaseInsensitivePath(std::string input)
+{
+    if (input == "/" || input == ".") {
+        return input;
+    }
+
+    const std::size_t input_len = input.length();
+
+    std::string out;
+    out.reserve(input_len);
+
+    std::vector<std::string> segments;
+    boost::split(segments, input, boost::is_any_of("/"));
+
+    size_t segments_traversed = 0;
+    for (const auto& segment : segments) {
+        if (segment.empty()) {
+            segments_traversed += 1;
+            continue;
+        }
+
+        if (segment == ".") {
+            if (!out.empty() && out.back() != '/') {
+                out += '/';
+            }
+            out += "./";
+            segments_traversed += 1;
+            continue;
+        }
+
+        if (out.empty() or out.back() != '/') {
+            out += '/';
+        }
```

Comment:
This is incorrect if path is relative, but doesn't start with a `./`, e.g. "/lr2skin/if2.lr2skin".

Reply:
7b7c5d65a42f2ab28fbc8ef4fe57d9ef773a375c